[chartmuseum] move from incubator to stable

jdolitsky opened the pull request:
Also, adding new Kubernetes org member @goruha to OWNERS

@sameersbn @prydonius this is a rehash of #3314 back in January. We have since removed the dependency on common chart.

jdolitsky committed:
[incubator/chartmuseum] update to latest stable image, add new params to values.yaml

jdolitsky committed:
Merge branch 'master' of github.com:kubernetes/charts

jdolitsky committed:
[incubator/chartmuseum] upgrade to 0.4.2

jdolitsky committed:
Merge branch 'master' of github.com:kubernetes/charts

jdolitsky committed:
[chartmuseum] move from incubator to stable

- Set "deprecated: true" in incubator Chart.yaml, bump patch (0.4.7)
- Add note to incubator README and NOTES that chart is deprecated
- New chart version in stable with minor version bump (0.5.0)
- Add new kubernetes org member @goruha to OWNERS
- Add Codefresh to list of maintainers

jdolitsky committed:
[chartmuseum] move from incubator to stable

- Set "deprecated: true" in incubator Chart.yaml, bump patch (0.4.7)
- Add note to incubator README and NOTES that chart is deprecated
- New chart version in stable with minor version bump (0.5.0)
- Add new kubernetes org member @goruha to OWNERS
- Add Codefresh to list of maintainers

jdolitsky committed:
Merge branch 'master' of github.com:jdolitsky/charts

jdolitsky committed:
[chartmuseum] fix maintainer names to match github orgs

jdolitsky committed:
[chartmuseum] incubator maintainer names to match stable

jdolitsky commented:
/assign @prydonius

sameersbn commented on stable/chartmuseum/README.md:
````diff
@@ -0,0 +1,350 @@
+## Installation
+
+```shell
+helm install --name my-chartmuseum -f custom.yaml incubator/chartmuseum
````
should be updated to `stable/chartmuseum` in the README

sameersbn commented on stable/chartmuseum/README.md:
````diff
@@ -0,0 +1,350 @@
+helm delete --purge my-chartmuseum
+```
+
+
````
Please remove the extra new line characters

sameersbn commented on stable/chartmuseum/templates/NOTES.txt:
```diff
@@ -0,0 +1,3 @@
+Thank you for installing {{ .Chart.Name }}.
+
+Your release is named {{ .Release.Name }}.
```
Can we have some additional notes on how the user can access the deployment or reach the API?

sameersbn commented on stable/chartmuseum/values.yaml:
```diff
@@ -0,0 +1,144 @@
+  nodePort:
+  annotations: {}
+
+resources:
```
Unless absolutely required we do not recommend enforcing resource requests/limits on charts. Should ideally be:

```
resource: {}
```

sameersbn commented on stable/chartmuseum/values.yaml:
```diff
@@ -0,0 +1,144 @@
+affinity: {}
+
+persistence:
+  Enabled: false
```
we should use consistent naming. Please use camelCase for the keys here as well

unguiculus replied:
Keys should start with a lower case character. I guess that's what you mean @sameersbn?

sameersbn replied:
yes. thats correct. the entire `persistence:` block uses TitleCase breaking consistency

unguiculus commented on stable/chartmuseum/templates/deployment.yaml:
```diff
@@ -0,0 +1,92 @@
+        - --storage-local-rootdir=/storage
+{{- end }}
+        ports:
+        - containerPort: {{ .Values.service.internalPort }}
```
Does it make sense to have the container port configurable? Just wondering. Also, you could give it a name and reference it by its name from service and probes.

unguiculus commented:
/assign

unguiculus commented:
There seems to be an issue with a recent change in CI scripts. Maintainers seem to always be required but we want them to be removed from deprecated charts.

See #4161.

prydonius commented on stable/chartmuseum/Chart.yaml:
```diff
@@ -0,0 +1,18 @@
+apiVersion: v1
+description: Helm Chart Repository with support for Amazon S3 and Google Cloud Storage
+name: chartmuseum
+version: 0.5.0
```
should this be a major version bump (1.0.0?). I think a minor version bump is okay for `0.` releases, but perhaps the major version would be more indicative of some of these proposed value changes.

scottrigby replied:
I was wondering something similar. If we are changing helm values YAML keys, then there's backwards incompatibility at those points, so that requires a semver MAJOR bump.

scottrigby commented on stable/chartmuseum/templates/secret.yaml:
```diff
@@ -0,0 +1,13 @@
+{{- if not (empty $value) }}
+  {{ $name }}: {{ $value | b64enc }}
+{{- end }}
+{{- end }}
```
Needs newline at EOF

jdolitsky committed:
Merge branch 'master' of github.com:kubernetes/charts

jdolitsky committed:
[chartmuseum] copy improvements from incubator chart

jdolitsky commented:
@prydonius @sameersbn @unguiculus @scottrigby we have addressed your comments.

@goruha made several changes in the incubator chart involving values etc., and that version has been set to 1.0.0. Now it will be deprecated, and set to 1.0.1, and the new stable version will be 1.1.0

jdolitsky committed:
[chartmuseum] upgrade to chartmuseum v0.5.1

jdolitsky commented:
@unguiculus -
> There seems to be an issue with a recent change in CI scripts. Maintainers seem to always be required but we want them to be removed from deprecated charts.

Are we blocked on merging this PR until that is fixed? Should I change incubator to `maintainers: []`?

jdolitsky committed:
Merge branch 'master' of github.com:kubernetes/charts

jdolitsky committed:
[chartmuseum] copy updates from incubator chart

jdolitsky committed:
Merge branch 'master' of github.com:kubernetes/charts

jdolitsky committed:
[chartmuseum] bump incubator version for deprecate

jdolitsky committed:
[chartmuseum] remove maintainers from incubator

unguiculus commented:
/lgtm
